broadinstitute/pearl
# JN-36 adding withdrawal functionality

This adds withdrawal capability to the admin tool.  Currently non-superusers can see the functionality, but only superusers can execute it. 
![image](https://github.com/broadinstitute/pearl/assets/2800795/0fcd36b3-c5e0-4cd6-8359-5f3808b07b0a)


 Withdrawal functionality has two competing concerns:
1. When an enrollee withdraws, they want their data to be totally gone, and we want to make sure it doesn't accidentally surface
2. For legal reasons, we need to keep a lot of their data (e.g. signed consents)

So, what we do is essentially turn the enrollee data into a json blob and park it in a `withdrawn_enrollee` table.  Then we hard-delete the enrollee record.  So we'll still have the enrollee data if, e.g., we need to produce a record of the signed consent.  But it won't be in a form that we have to worry about it accidentally getting included in a query.  (we had originally planned on doing a softer delete via the "withdrawn" boolean on enrollee, but on further reflection, that brought too much risk of accidental data inclusion)

This also adds an extremely trivial study environment stats endpoint (not even linked from anywhere yet) for viewing withdrawn enrollee counts

TO REVIEW:
1. start with the schema change
2. Look at the withdrawEnrollee method in WithdrawnEnrolleeService to see the withdrawal process
3. Look at WithdrawnEnrolleeServiceTests and PopulateOurhealthTests to see the consequences of withdrawal. 
4. everything else

TO TEST:
1. restart admin tool
5. run `./scripts/populate_portal.sh ourhealth`
6. go to `https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/stats`
7. notice there is now a withdrawn enrollee
8. go to `https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/participants`
9. confirm the withdrawn enrollee "Will drawn" does not appear
10. pick an existing participant from the list
11. Follow the https://broadworkbench.atlassian.net/wiki/spaces/PEARL/pages/2751922177/Enrollee+withdrawal wiki (although please do not delete them from the Azure tenant, or if you do, restore them afterwards)
